Add automation trigger WooCommerce Booking status change

lysyjan opened the pull request:
## Description

_N/A_

## Code review notes

- The trigger icon should be changed later

## QA notes

_N/A_

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/978

## Linked tickets

STOMAIL-7377

## After-merge notes

_N/A_

## Tasks

- [ ] I have added a changelog entry (`./do changelog:add --type=<type> --description=<description>`)
- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:stomail-7377-trigger-booking-status-change)

_The latest successful build from `stomail-7377-trigger-booking-status-change` will be used. If none is available, the link won't work._

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - WooCommerce Bookings integration added to Automations (visible when Bookings is active).
  - New "Booking status changed" trigger with icon, searchable keywords, and premium upsell edit modal.
  - Editor now initializes and detects Bookings availability to surface relevant options.

- **Automation**
  - Improved subject handling so customer-based workflows map to subscriber subjects when applicable.

- **Analytics**
  - Plugin activation reporting now includes WooCommerce Bookings.

- **Chores**
  - Frontend flag added to expose Bookings availability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lysyjan committed:
Add WooCommerce Bookings helper class and update DI container configuration

STOMAIL-7377

lysyjan committed:
Add support for detecting WooCommerce Bookings plugin activation

STOMAIL-7377

lysyjan committed:
Add methods for retrieving WooCommerce booking statuses and individual bookings

STOMAIL-7377

lysyjan committed:
Add trigger for WooCommerce Booking status changes and initialize integration

STOMAIL-7377

lysyjan committed:
Add stub class for WooCommerce Bookings in PHPStan analysis

STOMAIL-7377

NeosinneR commented on mailpoet/tasks/phpstan/custom-stubs.php:
```diff
@@ -59,6 +59,26 @@ public function get_date(string $dateType, string $timeZone = 'gmt') {
   }
 
+  if (!class_exists(\WC_Booking::class)) {
+    class WC_Booking extends WC_Order {
```
This is not correct. `WC_Booking` is a [not related to order class](https://github.com/woocommerce/woocommerce-bookings/blob/48ca281e7060804d3f39a55ea5a614eb74bf9257/includes/data-objects/class-wc-booking.php#L6) but [extends `WC_Bookings_Data`](https://github.com/woocommerce/woocommerce-bookings/blob/48ca281e7060804d3f39a55ea5a614eb74bf9257/includes/compatibility/abstract-wc-bookings-data.php), which in turn [extends `WC_Data`](https://github.com/woocommerce/woocommerce/blob/71e62b893c1943cef71b3180d723a96019dce6a3/plugins/woocommerce/includes/abstracts/abstract-wc-data.php).

NeosinneR reviewed (changes requested):
Looks good, @lysyjan. I have a small change request in the comment below regarding stubs.

And one more thing, that wasn't in specs, but could you please add tracking of WC Bookings plugin to [`Reporter`](https://github.com/mailpoet/mailpoet/blob/8af83a80a61fcc41f59ddacf6cf0b0e1de7b7634/mailpoet/lib/Analytics/Reporter.php#L210-L215)?

lysyjan committed:
Fix stub for WC_Booking

There was used an incorrect parent that caused some missing methods for PHPStan.
STOMAIL-7377

lysyjan committed:
Add reporting about active WooCommerce Bookings plugin

STOMAIL-7377

lysyjan committed:
Add CustomerToSubscriberSubjectTransformer for MailPoet integration

STOMAIL-7377

NeosinneR reviewed (approved):
Thanks for the changes, @lysyjan. I tested status change on a booking with or without order, as well as some of the booking filters and all worked as expected.

Aschepikov reviewed (approved):
GJ!
